### DISCUSSION TITLE
Splitting Projections and Aggregates Across Applications

### DISCUSSION THREAD

**Discussion by b-barry** [spatie/laravel-event-sourcing#480] [Q&A] [answered]
I'm considering ways to optimize our event sourcing setup with spatie/laravel-event-sourcing and have a question about splitting components:

- Is it possible to split different projections into separate applications? For example, having one instance per related group of projections to scale each independently.
- Similarly, can we distribute aggregate roots across different applications? What would this look like in practice?
- If feasible, what are the key considerations for implementing this kind of split? How would event routing and consistency be maintained?
- Are there any performance benefits or drawbacks to this approach compared to a centralized setup?

The goal is to allow for more targeted scaling and resource allocation.

**Comment by inmanturbo** [comment, marked as answer]
I have done this or something similar using composer and dependencies and service providers. You can put for instance an aggregate in one package, then put an aggregate partial and/or a projector that "subscribes" to certain events handled by the aggregate in another package, and register the projector in the package's service provider. Simply install both packages into your Laravel app. 

If you want them truly running entirely separate (as in possible on another app server or droplet, container etc) then that would require a lot of additional set up on your part, with no help from the package.
